File: src/main/java/com/openpms/server/app/dto/LoginBody.java
package com.openpms.server.app.dto;

import com.alibaba.fastjson2.annotation.JSONField;
import com.alibaba.fastjson2.annotation.JSONType;
import lombok.Data;

@Data
@JSONType(schema = """
    {
        "required": ["email", "password"]
    }"""
)
public class LoginBody {
    @JSONField(schema = "{'format':'email'}")
    private String email;
    @JSONField(schema = "{'minLength': '8'}")
    private String password;
}

File: src/main/kotlin/com/creator/server/app/dto/LoginBody.kt
package com.creator.server.app.dto

import com.alibaba.fastjson2.annotation.JSONField
import com.alibaba.fastjson2.annotation.JSONType

@JSONType(
    schema = """
    {
        "required": ["email", "password"]
    }
    """
)
data class LoginBody(
    @JSONField(schema = "{'format':'email'}")
    val email: String? = null,
    @JSONField(schema = "{'minLength': '8'}")
    val password: String? = null
)
